docs/LifeOfAnInstruction.md

```diff
@@ -0,0 +1,202 @@
+# Machine code to bitcode: the life of an instruction
+
+This document describes how machine code instructions are lifted into LLVM bitcode. It should provide a detailed, inside-out overview of how McSema performs binary translation. This document omits some important steps, so consulting the [Navigating the code](NagivatingTheCode.md) document will be helpful.
+
+### Running example
+
+This document will use the instructions in the following basic block as a running example.
+
+![Sample basic block](images/instruction_life_block.png)
+
+## Step 1: CFG protocol buffer representation
+
+The first step to lifting involves getting the machine code instructions of some binary file into a format the McSema understands. We use a common and simple format so that McSema doesn't need to understand the many executable file formats ([ELF](https://en.wikipedia.org/wiki/Executable_and_Linkable_Format), [Mach-O](https://en.wikipedia.org/wiki/Mach-O), [PE](https://en.wikipedia.org/wiki/Portable_Executable), etc.).
```

Comment:
No real need to talk about executable formats; just say that mcsema uses an external tool, currently IDA, to identify instructions, references, and data fields in various binaries.